### PR TITLE
Added histogramdd function

### PR DESCRIPTION
**Pull request by nomeshkumar**
- [ ] #12703

### CODE REVIEW

File: ivy/functional/backends/numpy/experimental/statistical.py [hunk 1]
```diff
@@ -5,6 +5,24 @@
 from ivy.func_wrapper import with_supported_dtypes
 from . import backend_version
 
+def histogramdd(
+        input: np.ndarray,
+        /,
+        *,
+        bins: Optional[Union[Tuple[int],int,Tuple[float]]] = 10,
```
Comment (hmahmood24) [inline ivy/functional/backends/numpy/experimental/statistical.py:12]: The type hinting for generic arguments that do not involve any native elements should be the same across each file

File: ivy/functional/backends/tensorflow/experimental/statistical.py [hunk 2]
```diff
@@ -6,6 +6,25 @@
 from ivy.func_wrapper import with_supported_dtypes
 from . import backend_version
 
+def histogramdd(
+        input: Union[tf.Tensor, tf.Variable],
```
Comment (hmahmood24) [inline ivy/functional/backends/tensorflow/experimental/statistical.py:10]: The input can only be a `tf.Tensor`

File: ivy/functional/backends/torch/experimental/statistical.py [hunk 3]
```diff
@@ -6,6 +6,23 @@
 from ivy.func_wrapper import with_unsupported_dtypes
 from . import backend_version
 
+def histogramdd(
+        input: torch.tensor,
+        /,
+        *,
+        bins: Optional[Union[Tuple[int],int,torch.Tensor]] = 10,
+        range: Optional[Union[Tuple[int], int]] = [0., 1., 0., 1.],
+        weights : torch.tensor,
```
Comment (hmahmood24) [inline ivy/functional/backends/torch/experimental/statistical.py:15]: You cannot define non-default arguments after default arguments. Please provide a default value for `weights`

File: ivy/functional/ivy/experimental/statistical.py [hunk 4]
```diff
@@ -9,6 +9,96 @@
 from ivy.utils.exceptions import handle_exceptions
 
 
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_nestable
+@handle_exceptions
+def histogramdd(
+        a: Union[ivy.Array, ivy.NativeArray],
```
Comment (hmahmood24) [inline ivy/functional/ivy/experimental/statistical.py:17]: Rename this to `input` as you've done in all the other files

File: ivy/functional/ivy/experimental/statistical.py [hunk 5]
```diff
@@ -9,6 +9,96 @@
...
+        if ``bins`` is an int, it defines the number of equal-width bins in the given
+        range.
+        if ``bins`` is an array, it defines a monotonically increasing array of bin
+        edges, including the rightmost edge, allowing for non-uniform bin widths.
+    range
+        the lower and upper range of the bins. The first element of the range must be
+        less than or equal to the second.
+    weights
+        each value in ``a`` only contributes its associated weight towards the bin count
+        (instead of 1). Must be of the same shape as a.
```
Comment (hmahmood24) [inline ivy/functional/ivy/experimental/statistical.py:44]: Change `a` to `input`

File: ivy/functional/ivy/experimental/statistical.py [hunk 6]
```diff
@@ -9,6 +9,96 @@
...
+        density: Optional[bool] = False
+) -> ivy.Array:
+    """
+    Compute the histogramdd of the NxD array ``a``.
+    .. note::
+        Given bins = [c0, ..., cK], defining intervals I0 = [c0, c1), I1 = [c1, c2),
+        ..., I_{K-1} = [c_{K-1}, cK].
+    Parameters
+    ----------
+    a
```
Comment (hmahmood24) [inline ivy/functional/ivy/experimental/statistical.py:32]: Change `a` to `input`

File: ivy/functional/backends/tensorflow/experimental/statistical.py [hunk 7]
```diff
@@ -6,6 +6,25 @@
...
+        /,
+        *,
+        bins: Optional[Union[Tuple[int],int,Tuple[float]]] = 10,
+        weights : Optional[Tuple[int]] = None,
+        range: Optional[Union[Tuple[int], int]] = [0., 1., 0., 1.],
+        density: Optional[bool] = False
+) -> Union[tf.Tensor, tf.Variable]:
```
Comment (hmahmood24) [inline ivy/functional/backends/tensorflow/experimental/statistical.py:17]: Kindly keep the return type consistent in all files. Remove `tf.Variable` and add `Tuple` in the beginning instead like you have done in other files.

File: ivy/functional/ivy/experimental/statistical.py [hunk 8]
```diff
@@ -9,6 +9,96 @@
...
+@handle_exceptions
+def histogramdd(
+        a: Union[ivy.Array, ivy.NativeArray],
+        /,
+        *,
+        bins: Optional[Union[Tuple[int], int,Tuple[float]]] = 10,
+        range: Optional[Tuple[float]] = None,
+        weights: Optional[Union[ivy.Array, ivy.NativeArray]] = None,
+        density: Optional[bool] = False
+) -> ivy.Array:
```
Comment (hmahmood24) [inline ivy/functional/ivy/experimental/statistical.py:24]: Kindly keep the return type consistent across all files.

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @nomeshkumar!
Thanks for an awesome contribution! I have left some comments and requested some changes. Kindly look into those. There are also a few points I want to mention that you overlooked in your implementation. 

1. You are supposed to add implementations in the `experimental` directories of not just all the Backend APIs and the Ivy Functional Experimental API, but also the **Array and Container** APIs as described [here](https://github.com/unifyai/ivy/issues/3856#issue-1356871035). You can look into [this](https://github.com/unifyai/ivy/pull/7462/files) PR for example as a reference so you know where else you need to add implementations of your function.
2. There are some linting and formatting checks that your PR is failing. Kindly look into those [here](https://github.com/unifyai/ivy/actions/runs/4521569029/jobs/7963383153?pr=13175) and fix them. I'm guessing you haven't installed `pre-commit` since it automatically runs all these checks whenever you commit your code and push it. I suggest you [set up](https://lets-unify.ai/ivy/overview/contributing/setting_up.html#pre-commit) `pre-commit` in your local environment and [run](https://lets-unify.ai/ivy/overview/deep_dive/formatting.html#pre-commit-hook) these linting and formatting checks because otherwise the CI tests will fail every time you submit a PR that has linting and formatting issues.
3. You have placed your test in the wrong directory. They are supposed to go in the `ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py` directory. I'm guessing you didn't run the tests locally because `pytest` would have thrown an error otherwise. Kindly always make sure that you run all your tests locally and ensure that they pass for ALL backends because otherwise the CI tests will fail when you submit a PR. 

Once you've made all the requested changes and ensured that the tests are running fine, kindly re-request a review from me. Good luck! 😊

**Comment by hmahmood24** [comment]
Also just a small point. Kindly ensure that your PRs gets linked to the issue(s) you are trying to resolve by commenting: `Close #<issue_number>` as explained [here](https://lets-unify.ai/ivy/overview/contributing/the_basics.html#todo-list-issues). This just helps in keeping track of tasks that have been carried out and correctly closing any issues that should be closed once the PR gets merged. Thank you! 😊

**Comment by nomeshkumar** [comment]
@hmahmood24 Thank you so much for your review, I'll fix these issues and let you know.

**Comment by nomeshkumar** [comment]
Close #12703

**Comment by nomeshkumar** [comment]
Hey @hmahmood24 , I have made all changes that you mentioned in comments except implementation of function in **Array and Container APIs**  because I couldn't find these folders in master branch

**Comment by hmahmood24** [comment]
> Hey @hmahmood24 , I have made all changes that you mentioned in comments except implementation of function in **Array and Container APIs** because I couldn't find these folders in master branch

Hey @nomeshkumar, The Array and Container APIs are located [here](https://github.com/unifyai/ivy/tree/master/ivy/data_classes). You'll need to implement your function in the `experimental` directories of both these modules as implemented in the reference PR I attached last time. 🙂

**Comment by nomeshkumar** [comment]
> > Hey @hmahmood24 , I have made all changes that you mentioned in comments except implementation of function in **Array and Container APIs** because I couldn't find these folders in master branch
> 
> Hey @nomeshkumar, The Array and Container APIs are located [here](https://github.com/unifyai/ivy/tree/master/ivy/data_classes). You'll need to implement your function in the `experimental` directories of both these modules as implemented in the reference PR I attached last time. 🙂

Okay thanks @hmahmood24

**Comment by nomeshkumar** [comment]
Close #12703

**Comment by nomeshkumar** [comment]
@hmahmood24 Kindly review the updates

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
I have reviewed the changes and left comments and changes for you to address. Again,

1. You have placed the test in the wrong directory. It will go in `ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py` file.
2. You have copied over the docstring from one file into all the other files. They should be modified according to each file/class they are being placed in. Please look at the docstrings of methods/functions in each file to get an idea of how to modify the docstrings in these files.
3. I don't think you have gone through my points [here](https://github.com/unifyai/ivy/pull/13175#pullrequestreview-1358000490). I can see plenty of linting/formatting issues with the PR. I also know that the tests would be failing since the test was not placed in the required file. Some of the functions had missing arguments as well. If you go over what I said [here](https://github.com/unifyai/ivy/pull/13175#pullrequestreview-1358000490), you need to fix all these formatting issues as well as make sure you run the test for **ALL** backends and make it pass so that we can go ahead and merge the PR.

Please run your tests locally using `pytest` as explained in my first comment at the top of this page as well as fix the formatting issues using `pre-commit` as mentioned in that same comment above. I won't be able to merge your PR if the tests aren't ran and verified to be passing. Good luck!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
